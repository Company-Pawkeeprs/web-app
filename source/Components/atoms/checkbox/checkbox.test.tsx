import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { describe, expect, it, vi } from 'vitest'
import Checkbox from './checkbox'

describe('Checkbox component (Unit)', () => {
    it('snapshot', () => {
        const { asFragment } = render(<Checkbox />)
        expect(asFragment()).toMatchSnapshot()
    })
    it('should render a checkbox', () => {
        render(<Checkbox />)
        const checkbox = screen.getByRole('checkbox')
        expect(checkbox).toBeInTheDocument()
    })

    it('should render a checkbox with custom class', () => {
        render(<Checkbox className="custom-checkbox" />)
        const checkbox = screen.getByRole('checkbox')
        expect(checkbox).toHaveClass('custom-checkbox')
    })

    it('should call onChange handler when clicked', async () => {
        const handleChange = vi.fn()
        render(<Checkbox onChange={handleChange} />)
        const checkbox = screen.getByRole('checkbox')
        await userEvent.click(checkbox)
        expect(handleChange).toHaveBeenCalled()
    })

    it('not should call onChange handler when clicked if disabled', async () => {
        const handleChange = vi.fn()
        render(<Checkbox onChange={handleChange} disabled />)
        const checkbox = screen.getByRole('checkbox')
        await userEvent.click(checkbox)
        expect(handleChange).not.toHaveBeenCalled()
    })

    it('should be checked when checked prop is true', () => {
        render(<Checkbox checked />)
        const checkbox = screen.getByRole('checkbox')
        expect(checkbox).toBeChecked()
    })

    it('should not be checked when checked prop is false', () => {
        render(<Checkbox checked={false} />)
        const checkbox = screen.getByRole('checkbox')
        expect(checkbox).not.toBeChecked()
    })

    it('should be disabled when disabled prop is true', () => {
        render(<Checkbox disabled />)
        const checkbox = screen.getByRole('checkbox')
        expect(checkbox).toBeDisabled()
    })

    it('should not be disabled when disabled prop is false', () => {
        render(<Checkbox disabled={false} />)
        const checkbox = screen.getByRole('checkbox')
        expect(checkbox).not.toBeDisabled()
    })
})
