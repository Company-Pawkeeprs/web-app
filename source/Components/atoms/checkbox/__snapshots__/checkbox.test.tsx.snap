// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkbox component (Unit) > snapshot 1`] = `
<DocumentFragment>
  <input
    class="rounded-sm duration-100 ease-in-out transition-shadow disabled:opacity-70 disabled:bg-gray-200 invalid:!border-red-500 invalid:focus:!border-gray-300 invalid:!shadow-outline-red invalid:!text-red-500 focus:outline-none font-sans dark:bg-[#292e33] dark:text-gray-200 border-primary-500 !w-4 h-4 px-0 py-0 mx-1 focus:!ring-0 border !border-primary-500 active:bg-secondary-500 text-primary-500"
    type="checkbox"
  />
</DocumentFragment>
`;
